### PR TITLE
TASK: Update dependencies of non-core Neos packages

### PR DESCRIPTION
**Pull request by mficzel**
Set the dependencies for packages that are maintained outside the neos core for the 4.3 base-distribution:

```
"neos/demo": "~5.0.0",
"neos/neos-ui": "~3.3",
"neos/seo": "~2.1",
"neos/redirecthandler-neosadapter": "~3.0",
"neos/redirecthandler-databasestorage": "~3.0",
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,15 +18,15 @@
...
         "neos/neos": "4.3.x-dev",
         "neos/nodetypes": "4.3.x-dev",
-        "neos/demo": "@dev",
         "neos/site-kickstarter": "4.3.x-dev",
 
-        "neos/neos-ui": "@dev",
+        "neos/demo": "~5.0.0",
+        "neos/neos-ui": "^3.3",
+        "neos/seo": "~2.1",
+        "neos/redirecthandler-neosadapter": "~3.0.0",
```
Comment (gerhard-boden) [inline composer.json]: Why only bugfix releases for the redirect plugins  but minors for neos /seo?
Also just bugfixes for neos/demo

**Comment by gerhard-boden** [comment]
left a comment in the code (not sure why it's not shown here)

File: composer.json (hunk 1 above)
Comment (mficzel) [reply composer.json]: @gerhard-boden streamligned the definition. Now only Neos.Demo uses a defined minor version. That makes sense so at least the visible neos you get is the same for the 4.3 branch.

**Review by gerhard-boden — APPROVED** [review]
now it looks logical too me thanks 👍

**Pull request merged**